docs/resources/softlayer_user.md

```diff
@@ -67,6 +120,16 @@ The following arguments are supported:
 * `last_name` | *string*
     * User's last name.
     * **Required**
+* `username` | *string*
+    * A name that uniquely identifies a user globally across all SoftLayer
+      logins. It is also the login userid. Once a user login is created,
+      it cannot be changed.
+    * You need to define `username` when your account is a SoftLayer 
+    credential account. In a IBMid account, it is generated by SoftLayer. 
+    For example, if an IBMid account number is `1234567` and email(IBMId) is 
+    `test@example.com`, `1234567_test@example.com` will be generated by
+    SoftLayer as `username`. 
+    * **Optional**
```

Comment:
I would add to line 132. `Optional for IBMid user accounts, Required otherwise.` and with the explanation on line 127 following.